### PR TITLE
Move level between stages in stage editor gui

### PR DESCRIPTION
**Pull request by davidsbailey**
# Description

Finishes [LP-757](https://codedotorg.atlassian.net/browse/LP-757)

Before, levels could be dragged to be reordered within a stage, but there was no way to move levels between stages using the GUI. This PR adds the following:
1. if a level is dragged to another stage, move it to the end of that stage
2. highlight the target stage while the level is being dragged, to give a clue that the level can be dropped there

collateral damage:
* replace `pageY` and `document.body.scrollTop` with `clientY` in the math for drag and drop, because `clientY` has the same frame of reference as `getBoundingClientRect`
* fix issues with dragging after scrolling by calling `forceUpdate` on drag start
* improve variable names

old UX:
busted-level-dragging

new UX:
new-level-dragging

## Testing story

I've added a unit test for the `moveLevelToStage` redux action, and manually tested dragging levels around between stages. There is no UI test, because we don't currently have the infrastructure to run UI tests against features which require levelbuilder mode. 

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/script-editor/LevelToken.jsx [hunk 1]
```diff
@@ -130,7 +131,7 @@ class LevelToken extends Component {
             style={Object.assign({}, styles.levelToken, {
               transform: `translate3d(0, ${y}px, 0) scale(${scale})`,
               boxShadow: `${color.shadow} 0 ${shadow}px ${shadow * 3}px`,
-              zIndex: this.props.drag ? 1000 : 500 - this.props.level.position
+              zIndex: draggedLevelPos ? 1000 : 500 - this.props.level.position
```
Comment (dmcavoy) [inline apps/src/lib/script-editor/LevelToken.jsx:134]: Why 1000 and 500?

Comment (davidsbailey) [reply apps/src/lib/script-editor/LevelToken.jsx:134]: just to be in front of everything else on the page. FWIW these numbers are unchanged in this PR, but always taking a second look at lines when touching them

**Review by dmcavoy — APPROVED** [review]
This looks good to me but I would wait for Brad to review as some of this stuff is newer to me

**Review by islemaster — APPROVED** [review]
Awesome!

**Pull request merged**